backend/app/readers/readers_instantiator.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from app.models.settings.settings import settings
+from app.readers.reader.reader import Reader
+from app.readers.reader.wikipedia_reader import WikipediaReader
+
+
+class ReadersInstantiator:
+    """
+    An Instantiator for the different Readers.
+    """
+
+    def __init__(self) -> None:
+        self.__record_types: frozenset[str] | None = self.__get_record_types()
+        self.__file_output_paths: list[Path] | None = self.__get_output_paths()
+
+    def __get_record_types(self) -> frozenset[str] | None:
+        """Retrieve a frozenset of record types from settings."""
+
+        if settings.record_types:
+            return settings.record_types
+        return None
+
+    def __get_output_paths(self) -> list[Path] | None:
+        """Retrieve output file names from settings and return a list of file output paths."""
+
+        if settings.output_file_names:
```

Comment:
This should be paths in the settings (may need a validator to convert str to paths). Then you don't need this method.

The general rule is to impose strong types at the boundaries rather than parsing lazily whenever possible.